auth/authentication.md

```diff
@@ -0,0 +1,61 @@
+# Authentication
+
+## User Flow
+
+User visits a new website that requires authentication from wallet.
+
+1. Website displays qrcode or deep link for auth
+2. User scans qrcode or redirects to wallet
+3. User approves prompt for auth in wallet
+4. User returns to website after prompt success
+5. Website is now authenticated with signed message
+
+User returns to a previously visted website before authentication expiry
+
+1. Website checks if user is authenticated from local storage
+2. Website is now authenticated with signed message
+
+User returns to a previously visted website after authentication expiry
+
+1. Website checks if user is authenticated from local storage
+2. Website sends new request to previously paired wallet
+3. User views notification and redirects to wallet
+4. User approves prompt for auth in wallet
+5. User returns to website after prompt success
+6. Website is now authenticated with signed message
+
+## Request Protocol
+
+A creates random symKey S for pairing topic.
+
+Pairing topic is the hash of symKey S.
+
+A generates keyPair X and generates response topic.
+
+Response topic is the hash of publicKey X.
+
+A will construct an authentication request.
+
+A encrypts request with symKey S.
+
+A publishes encrypted request to topic.
+
+A starts polling messages on response topic.
+
+B subscribes to pairing topic from provided URI
+
+B receives request and decrypts it with symKey S.
+
+B constructs message to be signed from request and signs it.
+
+B generates keyPair Y and generates shared symKey R.
+
+B encrypts response with symKey R as type 1 envelope.
+
```

Comment:
Looks like line how response topic is created is missing.

`Response topic is the hash of pubKey Y.`

Is that correct?

Reply:
It's mentioned on Line 35

Might need to restructure how we write these protocol flows